### DISCUSSION TITLE
Prerender with Nuxt and Vercel

### DISCUSSION THREAD

**Discussion by DidoMarchet** [vercel/vercel#6678] [Help] [answered]
Hi guys,
I would like to find some doc how to build a project with Nuxt in spa mode.

I need to setup the Prerender with Vercel.

If you have any advice let me know!

Thanks in advance and kind regards,

Davide

**Comment by leerob** [comment, marked as answer]
> Deprecated: please use ssr: false instead of mode: spa

Is there a reason you're wanting to not use SSR with Nuxt? Would it make more sense to use vanilla Vue instead? https://nuxtjs.org/docs/2.x/configuration-glossary/configuration-ssr

This should work out of the box on Vercel with your Nuxt project either way 😄

**Reply by DidoMarchet** [reply]
Hi @leerob 
and thanks for answering!

I'm sorry, I said SPA mode but I mean `ssr: False` (only client actually).

If I use SSR with Static generation, the dynamic pages takes long time to be rendered (I have 15k pages circa).
If I use SSR with Static generation, avoiding render dynamic pages, crawlers doesn't recognize the content.
Using `ssr: false` (only client ) I have the same issue above.

If I use `ssr: true` and `target: server` I need to setup a Node server to render the page and use `build` command and not `generate`.

Currently I use Netlify and they have this functionality: [https://docs.netlify.com/site-deploys/post-processing/prerendering/](https://docs.netlify.com/site-deploys/post-processing/prerendering/). This is perfect because let crawler like Facebook to crawl the content.

I want to know If Vercel has something similar because I would like to try and move to Vercel in fact.

Thanks in advance and regards,

Davide

**Reply by leerob** [reply]
> If I use SSR with Static generation, the dynamic pages takes long time to be rendered (I have 15k pages circa).
> If I use SSR with Static generation, avoiding render dynamic pages, crawlers doesn't recognize the content.

This isn't making sense for me. Server-side rendering and Static-site Generation are separate things, which is also different from client-side rendering (ssr: false).

When crawling, search engines do the rendering of the page themselves to determine the structure and content of the page (separate from the browser). Running JavaScript to generate the content is costly for search engines - running an entire headless Chrome and doing a full client rendering for every page of the site just doesn't make sense. This is why the "prerendering" mentioned above isn't a valid substitute for actual pre-rendered content from the server. 

Thus, if you use Nuxt by default, using SSR (aka serving pre-rendered content from the server), you won't have this issue.

**Reply by DidoMarchet** [reply]
Hi and thanks again.
You said exactly what I mean but my english is a gap.

Then Vercel pre-rendered content from the server and It is not only a CDN.

Thanks for your time and kind regards,

Davide

**Reply by leerob** [reply]
Happy to help 😄
